kuma/api/v1/search/__init__.py

```diff
@@ -67,7 +74,25 @@ def search(request, locale=None):
         params,
         make_suggestions=make_suggestions,
     )
-    return JsonResponse(results)
+    response = JsonResponse(results)
+
+    # The reasoning for caching is that most of the time, the seaches people make
+    # are short and often stand a high chance of being reused by other users
+    # in the CDN.
+    # The worst that can happen is that we fill up the CDN with cached responses
+    # that end up being stored there and never reused by another user.
+    # We could consider only bothering with this based on looking at the parameters.
+    # For example, if someone made a search with "complex parameters" we could skip
+    # cache-control because it'll just be a waste to store it (CDN and client).
+    # The reasoning for not using a "shared" cache-control, i.e. `s-max-age` is
+    # because the cache-control seconds we intend to set are appropriate for both
+    # the client and the CDN. If the value set is 3600 seconds, that means that
+    # clients might potentially re-use their own browser cache if they trigger
+    # a repeated search. And it's an appropriate number for the CDN too.
+    # For more info about how our search patterns behave,
+    # see https://github.com/mdn/kuma/issues/7799
+    patch_cache_control(response, public=True, max_age=SEARCH_CACHE_CONTROL_MAX_AGE)
```

Comment:
This might cause confusion when you're doing local development. E.g. localhost.org:8000/api/v1/search 
What we can do is something like:
```suggestion
    patch_cache_control(
        response, public=True, 
        max_age=0 if settings.DEBUG else SEARCH_CACHE_CONTROL_MAX_AGE
    )
```